fuzzers/main.py

```diff
@@ -0,0 +1,642 @@
+import atexit
+from collections import defaultdict
+import datetime
+import google
+import google.cloud.storage as gcs
+from http.server import BaseHTTPRequestHandler, HTTPServer
+import inotify.adapters
+import os
+import pathlib
+import prometheus_client
+import random
+import signal
+import subprocess
+import sys
+import threading
+import time
+import toml
+import typing
+import uuid
+
+from lib import config
+from workers import utils
+
+#WORKDIR = utils.WORKDIR
+#REPO_URL = utils.REPO_URL
+WORKDIR = pathlib.Path('/tmp')
+REPO_URL = 'https://github.com/Ekleog/nearcore'
+def connect_to_gcs() -> gcs.Client:
+    """Setup the environment to have gsutils work, and return a connection to GCS"""
+    #subprocess.run(
+    #    ['gcloud', 'auth', 'activate-service-account', '--key-file', GCS_CREDENTIALS_FILE],
+    #    check = True,
+    #)
+    #return gcs.Client.from_service_account_json(str(GCS_CREDENTIALS_FILE))
+    return gcs.Client(project = 'near-nayduck')
```

Comment:
Have you looked into using `workers.blobs`? In particular, I would want to take a look at using machine credentials which would change how authentication works and it’d be great if we did that in only one location.

Reply:
I have not yet, but anyway we'll have to have something specific here in order to make `gsutil rsync` work, and so there's no chance of making that work on Azure. So I'm not sure it'd be very helpful to merge with `workers.blobs`? Or, put another way, maybe it'd make sense to just refactor the GCS login process to a helper function in `lib` that'd be called by both places, that'd basically be `connect_to_gcs`?

Reply:
Yeah, we don’t care about Azure. I’ll send a change removing support for it.